core/sr-primitives/src/generic/unchecked_mortal_compact_extrinsic.rs

```diff
@@ -24,48 +24,63 @@ use runtime_io::blake2_256;
 use crate::codec::{Decode, Encode, Input, Compact};
 use crate::traits::{self, Member, SimpleArithmetic, MaybeDisplay, CurrentHeight, BlockNumberToHash,
 	Lookup, Checkable, Extrinsic, SaturatedConversion};
-use super::{CheckedExtrinsic, Era};
+use super::{CheckedExtrinsic, Era, Tip, Tippable};
 
 const TRANSACTION_VERSION: u8 = 1;
 
 /// A extrinsic right from the external world. This is unchecked and so
 /// can contain a signature.
 #[derive(PartialEq, Eq, Clone)]
-pub struct UncheckedMortalCompactExtrinsic<Address, Index, Call, Signature> {
+pub struct UncheckedMortalCompactExtrinsic<Address, Index, Call, Signature, Balance> {
 	/// The signature, address, number of extrinsics have come before from
 	/// the same signer and an era describing the longevity of this transaction,
 	/// if this is a signed extrinsic.
 	pub signature: Option<(Address, Signature, Compact<Index>, Era)>,
 	/// The function that should be called.
 	pub function: Call,
+	/// The tip for this transaction
+	pub tip: Tip<Balance>,
```

Comment:
You don't seem to modify `fn check` function and it's strictly required that `tip` is part of the payload we are signing. Otherwise you can change it arbitrarly and the signature would still be valid.

Reply:
right -- added to the todo list up top.

Reply:
cc @jacogr formats are changing

Reply:
As Jaco noticed we could bump version here.

Reply:
check is modified but I will keep the conversation open as a reminder that the extrinsic format has changed slightly.